### PR TITLE
AWS Cognito trigger + Customized Authentication UI

### PR DESCRIPTION
**Pull request by c3duan**
### Administrative Info
Monday Board ID: 853606148
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added and tested Cognito Trigger that inserts user metadata into DynamoDB table when a new user is created in Cognito user pool
- Modified missed Schema changes from last PR
- Customized Amplify Auth UI to look closer to Figma
- Added AppState Context to help with auth -> onboarding -> app flow
- Made frontend filename more consistent

### Testing
How did you confirm your changes worked? 
- Tested on AWS IAM that Cognito trigger works
- Tested Customized Amplify Auth UI on both Android and iOS

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Cognito Trigger |
| --- |
| GraphQL Query Works |
| Screen Shot 2021-04-04 at 11 41 47 PM |
| Item Inserted into DynamoDB Table |
| Screen Shot 2021-04-04 at 11 42 40 PM |
| CloudWatch Logs Showing Success |
| Screen Shot 2021-04-04 at 11 43 21 PM |

| Authentication UI |
| --- |
| iOS |
| Simulator Screen Shot - iPhone 12 Pro Max - 2021-04-06 at 12 01 35 | 
| Simulator Screen Shot - iPhone 12 Pro Max - 2021-04-06 at 12 01 47 |
| Android |
| Screenshot_20210406_120631 |
| Screenshot_20210406_120743 |

### CODE REVIEW

File: src/components/AuthContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Auth.configure({ mandatorySignIn: true });
+
+function App(): JSX.Element | null {
+  const { appState, setAppState } = useContext(AppContext);
+
+  useEffect(() => {
+    if (appState === 'Auth') {
+      setAppState('Onboarding'); // TODO: use datastore to check whether this user has at least one home
+    }
```
Comment (TheHiddenLayer) [inline src/components/AuthContainer.tsx:18]: ok.

File: src/screens/SplashScreen.tsx [hunk 2]
```diff
@@ -2,9 +2,9 @@ import React from 'react';
 import { View } from 'react-native';
 import Logo from '../assets/logo';
 
-import SplashScreenStyles from '../styles/splash-screen-styles';
+import SplashScreenStyles from '../styles/SplashScreenStyles';
```
Comment (TheHiddenLayer) [inline src/screens/SplashScreen.tsx:5]: good catch

**Review by TheHiddenLayer — APPROVED** [review]
Love the screenshots. Changes approved.

**Review by bobbytatum27 — APPROVED** [review]
Great work Arnold! Thanks for configuring the UI, too. Everything looks good to me.

**Pull request merged**